src/examples/algorithms/AnomalyDetection.cpp

```diff
@@ -0,0 +1,186 @@
+#include <iostream>
+#include <fstream> // for file output
+#include <vector>
+#include <string>
+#include <cstdlib>
+
+#include <csv.h> // external CSV parser
+
+#include "nupic/encoders/ScalarEncoder.hpp"
+#include "nupic/algorithms/SpatialPooler.hpp"
+#include "nupic/algorithms/Cells4.hpp"
+#include "nupic/algorithms/Anomaly.hpp"
+#include "nupic/os/Timer.hpp"
+#include "nupic/utils/VectorHelpers.hpp"
+
+using namespace std; // generic namespaces are included here
+using namespace nupic;
+using namespace nupic::utils;
+using namespace nupic::algorithms::spatial_pooler;
+using namespace nupic::algorithms::Cells4;
+using namespace nupic::algorithms::anomaly;
+
+const int DEBUG_LEVEL =1; //0=no debug (also disabled timer), ..
+
+class AnomalyDetection
```

Comment:
I would do this example differently:

- Use the network api instead of directly instantiating the algorithmic components
- See the [ScalarSensor](https://github.com/numenta/nupic.core/blob/master/src/nupic/encoders/ScalarSensor.hpp) that @mrcslws created for getting data into the network
- Use helper functions for things like contructing the network but don't define a helper class
- Keep the `main` function mostly as is - that looks great